File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0, maximum = nums[0];
        for(int num : nums){
            sum += num;
            maximum = max(maximum, sum);
            if(sum < 0)
                sum = 0;
        }
    
        return maximum;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.py
from math import *
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum = 0
        maxsum = nums[0]
        for x in range (len(nums)):
            sum = sum + nums[x]
            if(sum > maxsum):
                maxsum = sum
            if(sum < 0):
                sum = 0
        return maxsum
            
        